server/src/main/java/org/elasticsearch/index/shard/RemoveCorruptedShardSegmentsCommand.java

```diff
@@ -0,0 +1,565 @@
+                ////////// Index
+                terminal.println("");
+                terminal.println("Opening Lucene index at " + indexPath);
+                terminal.println("");
+                try {
+                    indexCleanStatus = removeCorruptedLuceneSegmentsAction.getCleanStatus(shardPath, indexDirectory,
+                        writeLock, printStream, verbose);
+                } catch (Exception e) {
+                    terminal.println(e.getMessage());
+                    throw e;
+                }
+
+                terminal.println("");
+                terminal.println(" >> Lucene index is " + indexCleanStatus.v1().getMessage() + " at " + indexPath);
+                terminal.println("");
+
+                ////////// Translog
+                // as translog relies on data stored in an index commit - to truncate translog - we have to have non unrecoverable index
+                if (indexCleanStatus.v1() != CleanStatus.UNRECOVERABLE) {
+                    terminal.println("");
+                    terminal.println("Opening translog at " + translogPath);
+                    terminal.println("");
+                    try {
+                        translogCleanStatus = truncateTranslogAction.getCleanStatus(shardPath, indexDirectory);
+                    } catch (Exception e) {
+                        terminal.println(e.getMessage());
+                        throw e;
+                    }
+
+                    terminal.println("");
+                    terminal.println(" >> Translog is " + translogCleanStatus.v1().getMessage() + " at " + translogPath);
+                    terminal.println("");
+                } else {
+                    translogCleanStatus = Tuple.tuple(CleanStatus.UNRECOVERABLE, null);
+                }
+
+                ////////// Drop corrupted parts
+                final CleanStatus indexStatus = indexCleanStatus.v1();
+                final CleanStatus translogStatus = translogCleanStatus.v1();
+                if (dryRun) {
+                    if (indexStatus == CleanStatus.CLEAN && translogStatus == CleanStatus.CLEAN) {
+                        terminal.println("Both Lucene index and traslog at " + shardPath.getDataPath() + " are clean.");
+                    } else {
+                        if (indexStatus != CleanStatus.CLEAN) {
+                            loseDataBanner(terminal, indexCleanStatus, dryRun);
+                        }
+
+                        if (translogStatus != CleanStatus.CLEAN) {
+                            loseDataBanner(terminal, translogCleanStatus, dryRun);
+                        }
+                    }
+                } else {
+
+                    if (indexStatus == CleanStatus.CLEAN && translogStatus == CleanStatus.CLEAN) {
+                        throw new ElasticsearchException("Both Lucene index and traslog at " + shardPath.getDataPath() + " are clean.");
```

Comment:
The use of `index` here is a bit confusing (vs Elasticsearch terminology) and there's a typo in `traslog`. How about something simpler:

> Shard does not seem to be corrupted.

Reply:
++